### DISCUSSION TITLE
Question About the Testing Endpoint

### DISCUSSION THREAD

**Discussion by cloneofghosts** [Pirate-Weather/pirateweather#93] [Q&A] [answered]
Was looking at the description on APIable for the Testing Endpoint it says:

> Want to follow along with development as it happens? This provides access to the testing (and very unstable for production use) endpoint

I know for the minor updates and some larger updates (#3) this is true but otherwise it seems like the updates are only ever pushed to the testing endpoint shortly before going live. Now I don't exactly know what your workflow is but I find it weird that the testing endpoint is advertised as having access to beta updates before they are production ready whereas currently it doesn't seem to be the case.

I am aware that some updates you cannot test on the testing endpoint and you'd probably want the update to at least be somewhat stable before testing. It would still be nice to have some sort of access to in-progress updates so we can help you test things and find bugs you didn't notice before going live.

I'm not sure if AWS has the ability to schedule the push to production since I was thinking that minor updates could be pushed up ~24h after going live on testing and for major updates ~7d after the final update is pushed up to test. If it isn't possible then ignore.

**Comment by alexander0042** [comment, marked as answer]
Good points here, and something I hadn't really thought about! You're definitely correct in noticing this- either the dev one goes live almost immediately, or I forget to push it and it stays there for a while, neither one is really ideal. Should be some more action on it as part of v2.0 though, since that will be pointed at a whole new function. I'm thinking something like "alpha", which will be very broken most of the time, "dev" which I like the 7 days thing, and "prod"

**Reply by cloneofghosts** [reply]
So with the 2.0 release this is something that will be utilized more? If an alpha endpoint is introduced will people who have access to the dev endpoint have access to the alpha endpoint or will it be restricted based on your API access? Just curious what the plan would be for that.

EDIT: I was thinking that those who have access to the dev endpoint currently could get access to the alpha endpoint and then the dev endpoint becomes available to everyone. Would make more sense since while it still won't be production ready it would be better to have as many people as possible available to test beta builds so you can fix more issues before launching the updates.

**Comment by cloneofghosts** [comment]
Since it seems like work on the 2.0 endpoint seems to be ramping up I was wondering what the status of this is as I haven't seen any alpha endpoints. In my last comment I mentioned that maybe alpha is restricted to people who pay for development access and dev becomes available to everyone?

**Reply by alexander0042** [reply]
Yea, I like that "Alpha/ Dev" breakdown.

The limiting factor before this goes live is that the API is two main parts, the ingest scripts and the response script. Step 1 to version 2.0 is reworking ingest to take the previous model results into account (finally fixing the daily high issue), but since they don't return data, I can't get the new endpoint setup yet. 

Getting excited for when it'll be able to go live though- it's looking like response times should get down to <1s, which would be a really great improvement on the current setup.
